### PR TITLE
updates common computation classes

### PR DESCRIPTION
**Pull request by Mathieu-Deharbe**
In reaction to https://github.com/gridsuite/security-analysis-server/pull/115

common computation classes need to be updated to preserve the harmonization.

----
Extra changes : 
* error messages are now truncated if too long (security added because of long request that could not be sent through rabbitmq)
* parameters message from : 
`parametersJson = objectMapper.writeValueAsString(objectMapper.createObjectNode().putPOJO(MESSAGE_ROOT_NAME, runContext.getParameters()));`
to : 
`parametersJson = objectMapper.writeValueAsString(runContext.getParameters());`
==> I dind't find any reason to keep the MESSAGE_ROOT_NAME (no other computation uses this => **should be checked : what does it change to have it as a root ?**

### CODE REVIEW

File: src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java [hunk 1]
```diff
@@ -169,27 +171,46 @@ public Consumer<Message<String>> consumeCancel() {
...
 
-        R context = resultContext.runContext;
-        String provider = context.getProvider();
+    /**
+     * if this computation needs extra report operations, override this function and do them here
+     */
+    protected void reportSpecificOperations(R runContext, Reporter reporter) { }
+
+    protected S run(R runContext, UUID resultUuid) throws Exception {
+        logOnRun(runContext);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java:189]: Remove logOnRun and enrichRunContext and add a preRun method

File: src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java [hunk 2]
```diff
@@ -169,27 +171,46 @@ public Consumer<Message<String>> consumeCancel() {
...
 
-        CompletableFuture<S> future = runAsync(network, context, provider, reporter, resultContext.getResultUuid());
+        Network network = getNetwork(runContext.getNetworkUuid(), runContext.getVariantId());
+        CompletableFuture<S> future = runAsync(network, runContext, provider, reporter, resultUuid);
 
-        S result = future == null ? null : observer.observeRun("run", context, future::get);
-        if (context.getReportContext().getReportId() != null) {
-            observer.observe("report.send", context, () -> reportService.sendReport(context.getReportContext().getReportId(), rootReporter.get()));
+        S result = future == null ? null : observer.observeRun("run", runContext, future::get);
+        if (runContext.getReportContext().getReportId() != null) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java:211]: Add a `postRun` method and remove reportSpecificOperations

File: src/main/java/org/gridsuite/loadflow/server/computation/utils/ContextUtils.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package org.gridsuite.loadflow.server.computation.utils;
+
+import com.powsybl.commons.PowsyblException;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * @author Thang PHAM <quyet-thang.pham at rte-france.com>
+ */
+public final class ContextUtils {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/loadflow/server/computation/utils/ContextUtils.java:15]: I'm not sure this is the right name 
It's not relative to a context but a message
`MessageUtils`

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractResultContext.java [hunk 4]
```diff
@@ -67,7 +66,9 @@ public Message<String> toMessage(ObjectMapper objectMapper) {
                 .setHeader(REPORT_UUID_HEADER, runContext.getReportContext().getReportId() != null ? runContext.getReportContext().getReportId().toString() : null)
                 .setHeader(REPORTER_ID_HEADER, runContext.getReportContext().getReportName())
                 .setHeader(REPORT_TYPE_HEADER, runContext.getReportContext().getReportType())
-                .setHeader(HEADER_LIMIT_REDUCTION, runContext.getLimitReduction() != null ? runContext.getLimitReduction().toString() : null)
+                .copyHeaders(getSpecificMsgHeaders())
             .build();
     }
+
+    public abstract Map<String, String> getSpecificMsgHeaders();
```
[the remark this answers is not in the document]
Comment (Mathieu-Deharbe) [reply src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractResultContext.java]: Done.

File: src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java [hunk 5]
```diff
@@ -53,32 +53,34 @@ public abstract class AbstractWorkerService<S, R extends AbstractComputationRunC
...
         this.reportService = reportService;
+        this.resultService = resultService;
         this.executionService = executionService;
         this.observer = observer;
         this.objectMapper = objectMapper;
-        this.resultRepository = resultRepository;
     }
 
-    protected Network getNetwork(AbstractComputationRunContext<P> runContext) {
+    public PreloadingStrategy getNetworkPreloadingStrategy() {
```
[the remark this answers is not in the document]
Comment (Mathieu-Deharbe) [reply src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java]: Done.

File: src/main/java/org/gridsuite/loadflow/server/computation/utils/ContextUtils.java (hunk 3 above)
Comment (Mathieu-Deharbe) [reply src/main/java/org/gridsuite/loadflow/server/computation/utils/ContextUtils.java]: Agreed. Changed. Is this ok for @thangqp the maker of this class ?

File: src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java [hunk 6]
```diff
@@ -124,13 +126,13 @@ public Consumer<Message<String>> consumeRun() {
...
 
-                Network network = getNetwork(resultContext.getRunContext());
-
-                S result = run(network, resultContext);
+                S result = run(resultContext.getRunContext(), resultContext.getResultUuid());
 
                 long nanoTime = System.nanoTime();
                 LOGGER.info("Just run in {}s", TimeUnit.NANOSECONDS.toSeconds(nanoTime - startTime.getAndSet(nanoTime)));
 
+                Network network = getNetwork(resultContext.getRunContext().getNetworkUuid(),
```
[the remark this answers is not in the document]
Comment (Mathieu-Deharbe) [reply src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java]: Done.

File: src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java (hunk 2 above)
Comment (Mathieu-Deharbe) [reply src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java:192]: Done.

File: src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java (hunk 1 above)
Comment (Mathieu-Deharbe) [reply src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java]: Done.

**Review by thangqp — APPROVED** [review]
Code OK, Test OK

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**